subscriber/common/sink/kafka.go

```diff
@@ -125,17 +129,17 @@ func (kp *KafkaPublisher) Cluster() string {
 	return kp.ClusterName
 }
 
-func (kp *KafkaPublisher) Insert(tableName string, shardID int32, columnNames []string, rows []client.Row,
-	updateModes ...memCom.ColumnUpdateMode) (int, error) {
+func (kp *KafkaPublisher) insert(tableName string, shardID int32, columnNames []string, rows []client.Row,
+	updateModes ...memCom.ColumnUpdateMode) {
 	kp.Scope.Gauge("batchSize").Update(float64(len(rows)))
 	saveStart := time.Now()
 	kp.ServiceConfig.Logger.Debug("saving", zap.Any("rows", rows))
 
 	bytes, numRows, err := kp.UpsertBatchBuilder.PrepareUpsertBatch(tableName, columnNames, updateModes, rows)
 	if err != nil {
 		kp.Scope.Counter("errors.insert").Inc(1)
-		return 0, utils.StackError(err, fmt.Sprintf("Failed to prepare rows in table %s, columns: %+v",
-			tableName, columnNames))
+		utils.StackError(err, "Failed to prepare rows in table %s, columns: %+v",
```

Comment:
error is not returned

Reply:
code changes, now replace single message with multiple messages, handle error differently

Reply:
maybe log the error